docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,175 @@
+= BSF Detector
+
+The BSF Detector lets you run arbitrary scripts using the Bean Scripting Framework to determine the availability of a service.
+This can be a script written in any of the languages BSF supports.
+The only requirement is that the script returns the string `OK` if it passes.
+
+Unlike the BSF Monitor, the BSF Detector only passes in the `map`, `ip_addr`, `svc_name`, and `results` variables as beans to the script.
+
+== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.bsf.BSFDetector`
+|===
+
+== Configuration and Use
+
+.Parameters for the BSF detector
+[options="header, autowidth"]
+
+|===
+| Parameter        | Description                                                                                            | Required | Default value
+| `fileName`       | The path to the file on the OpenNMS server which contains the BSF-compatible script to run.            | optional |
+| `langClass`      | The short name of the language the script is written in. Choices: Jython, BeanShell, Groovy, etc.     | optional | 
```

Comment:
```suggestion
| `langClass`      | The short name of the language the script is written in. Choices: `Jython`, `BeanShell`, `Groovy`, etc.     | optional | 
```

Reply:
It seems like the options you can use are lower case, right?

Reply:
Never saw upper case options TBH.